### PR TITLE
Add x-kubernetes-validations cost limit integration test

### PR DESCRIPTION
**Pull request by DangerOnTheRanger**
This PR adds an integration test to ensure that CustomResource creation fails if an x-kubernetes-validations rule is included that exceeds the runtime cost limit. While this PR was created as a part of feedback on #110576, it can be merged separately/before that PR. 

#### What type of PR is this?

/kind feature

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
Please ask the reviewers of the kep and the other pr to have a look.

**Comment by cici37** [comment]
/lgtm
We added the similar e2e test in https://github.com/kubernetes/kubernetes/pull/110576. Since the e2e test hasn't been added into conformance test suite, adding it into integration could improve our test coverage.

/assign @liggitt 
Would you mind to take a look when have time? Thank you 😊

File: test/integration/apiserver/crd_validation_expressions_test.go [hunk 1]
```diff
@@ -301,6 +301,36 @@ func TestCustomResourceValidators(t *testing.T) {
...
+			"apiVersion": gvr.Group + "/" + gvr.Version,
+			"kind":       crd.Spec.Names.Kind,
+			"metadata": map[string]interface{}{
+				"name": name1,
+			},
+			"spec": map[string]interface{}{
+				"list": genLargeArray(725, 20),
+			},
+		}}
+		_, err = crClient.Create(context.TODO(), cr, metav1.CreateOptions{})
```
Comment (liggitt) [inline test/integration/apiserver/crd_validation_expressions_test.go:327]: non-blocking, but I'm curious how long this create call takes before it fails (to see if our runtime protections are actually keeping the request handling time short)

Comment (DangerOnTheRanger) [reply test/integration/apiserver/crd_validation_expressions_test.go:327]: For what it's worth, averaged over 5 runs on my machine it takes about 539ms before it fails.

**Pull request merged**